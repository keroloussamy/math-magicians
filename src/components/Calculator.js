import React from 'react';
import './Calculator.css';

// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends React.Component {
  render() {
    return (
      <table className="calculator">
        <tbody>
          <tr>
            <th colSpan="4" className="th-result"><p className="calculator-result">0</p></th>
          </tr>
          <tr>
            <th className="w-25"><button type="button" className="btn-Gray">AC</button></th>
            <th className="w-25"><button type="button" className="btn-Gray">+/-</button></th>
            <th className="w-25"><button type="button" className="btn-Gray">%</button></th>
            <th className="w-25"><button type="button" className="btn-Orange">รท</button></th>
          </tr>
          <tr>
            <th><button type="button" className="btn-Gray">7</button></th>
            <th><button type="button" className="btn-Gray">8</button></th>
            <th><button type="button" className="btn-Gray">9</button></th>
            <th><button type="button" className="btn-Orange">x</button></th>
          </tr>
          <tr>
            <th><button type="button" className="btn-Gray">4</button></th>
            <th><button type="button" className="btn-Gray">5</button></th>
            <th><button type="button" className="btn-Gray">6</button></th>
            <th><button type="button" className="btn-Orange">+</button></th>
          </tr>
          <tr>
            <th><button type="button" className="btn-Gray">1</button></th>
            <th><button type="button" className="btn-Gray">2</button></th>
            <th><button type="button" className="btn-Gray">3</button></th>
            <th><button type="button" className="btn-Orange">-</button></th>
          </tr>
          <tr>
            <th colSpan="2"><button type="button" className="btn-Gray">0</button></th>
            <th><button type="button" className="btn-Gray">.</button></th>
            <th><button type="button" className="btn-Orange">=</button></th>
          </tr>
        </tbody>
      </table>
    );
  }
}

export default Calculator;
