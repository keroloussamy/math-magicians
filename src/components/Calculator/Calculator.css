.btn-Orange {
  background-color: #f5913e;
}

.btn-Gray {
  background-color: #e0e0e0;
}

.calculator-result {
  text-align: right;
  margin: 5px;
  height: 35px;
  font-size: 30px;
}

.calculator {
  margin-left: auto;
  margin-right: 80px;
  background-color: #858693;
  width: 30%;
  min-width: 300px;
  height: 500px;
}

button {
  width: 100%;
  height: 100%;
  font-size: 20px;
}

.th-result {
  height: 0;
}

th,
td {
  padding: 0;
}

.w-25 {
  width: 25%;
}

.calculator-container {
  margin-top: 10px;
  display: flex;
}

.calculator-container h2 {
  padding-left: 80px;
}
