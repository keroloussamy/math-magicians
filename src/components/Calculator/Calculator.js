import React, { useState } from 'react';
import './Calculator.css';
import CalcButton from '../CalcButton/CalcButton';
import calculate from '../../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({});

  const onClickHandlerFun = (e) => {
    const result = calculate(state, e.target.textContent);
    setState(result);
  };

  const { next, total } = state;
  return (
    <table className="calculator">
      <tbody>
        <tr>
          <th colSpan="4" className="th-result"><p className="calculator-result">{next ?? total ?? 0}</p></th>
        </tr>
        <tr>
          <th className="w-25"><CalcButton className="btn-Gray" name="AC" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th className="w-25"><CalcButton className="btn-Gray" name="+/-" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th className="w-25"><CalcButton className="btn-Gray" name="%" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th className="w-25"><CalcButton className="btn-Orange" name="รท" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
        </tr>
        <tr>
          <th><CalcButton className="btn-Gray" name="7" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th><CalcButton className="btn-Gray" name="8" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th><CalcButton className="btn-Gray" name="9" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th><CalcButton className="btn-Orange" name="x" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
        </tr>
        <tr>
          <th><CalcButton className="btn-Gray" name="4" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th><CalcButton className="btn-Gray" name="5" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th><CalcButton className="btn-Gray" name="6" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th><CalcButton className="btn-Orange" name="+" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
        </tr>
        <tr>
          <th><CalcButton className="btn-Gray" name="1" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th><CalcButton className="btn-Gray" name="2" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th><CalcButton className="btn-Gray" name="3" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th><CalcButton className="btn-Orange" name="-" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
        </tr>
        <tr>
          <th colSpan="2"><CalcButton className="btn-Gray" name="0" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th><CalcButton className="btn-Gray" name="." onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
          <th><CalcButton className="btn-Orange" name="=" onClickHandler={onClickHandlerFun} aria-label="Mute volume" /></th>
        </tr>
      </tbody>
    </table>
  );
};

export default Calculator;
