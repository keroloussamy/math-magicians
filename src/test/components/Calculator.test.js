import React from 'react';
import { render } from '@testing-library/react';
import Calculator from '../../components/Calculator/Calculator';

it('Calculator component', () => {
  const tree = render(<Calculator />);
  expect(tree).toMatchSnapshot();
});
