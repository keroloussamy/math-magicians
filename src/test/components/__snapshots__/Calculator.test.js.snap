// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculator-container"
      >
        <h2>
          Lets Do Some Math !!
        </h2>
        <table
          class="calculator"
        >
          <tbody>
            <tr>
              <th
                class="th-result"
                colspan="4"
              >
                <p
                  class="calculator-result"
                  data-testid="result"
                >
                  0
                </p>
              </th>
            </tr>
            <tr>
              <th
                class="w-25"
              >
                <button
                  class="btn-Gray"
                  type="button"
                >
                  AC
                </button>
              </th>
              <th
                class="w-25"
              >
                <button
                  class="btn-Gray"
                  type="button"
                >
                  +/-
                </button>
              </th>
              <th
                class="w-25"
              >
                <button
                  class="btn-Gray"
                  type="button"
                >
                  %
                </button>
              </th>
              <th
                class="w-25"
              >
                <button
                  class="btn-Orange"
                  type="button"
                >
                  รท
                </button>
              </th>
            </tr>
            <tr>
              <th>
                <button
                  class="btn-Gray"
                  type="button"
                >
                  7
                </button>
              </th>
              <th>
                <button
                  class="btn-Gray"
                  type="button"
                >
                  8
                </button>
              </th>
              <th>
                <button
                  class="btn-Gray"
                  type="button"
                >
                  9
                </button>
              </th>
              <th>
                <button
                  class="btn-Orange"
                  type="button"
                >
                  x
                </button>
              </th>
            </tr>
            <tr>
              <th>
                <button
                  class="btn-Gray"
                  type="button"
                >
                  4
                </button>
              </th>
              <th>
                <button
                  class="btn-Gray"
                  type="button"
                >
                  5
                </button>
              </th>
              <th>
                <button
                  class="btn-Gray"
                  type="button"
                >
                  6
                </button>
              </th>
              <th>
                <button
                  class="btn-Orange"
                  type="button"
                >
                  +
                </button>
              </th>
            </tr>
            <tr>
              <th>
                <button
                  class="btn-Gray"
                  type="button"
                >
                  1
                </button>
              </th>
              <th>
                <button
                  class="btn-Gray"
                  type="button"
                >
                  2
                </button>
              </th>
              <th>
                <button
                  class="btn-Gray"
                  type="button"
                >
                  3
                </button>
              </th>
              <th>
                <button
                  class="btn-Orange"
                  type="button"
                >
                  -
                </button>
              </th>
            </tr>
            <tr>
              <th
                colspan="2"
              >
                <button
                  class="btn-Gray"
                  type="button"
                >
                  0
                </button>
              </th>
              <th>
                <button
                  class="btn-Gray"
                  type="button"
                >
                  .
                </button>
              </th>
              <th>
                <button
                  class="btn-Orange"
                  type="button"
                >
                  =
                </button>
              </th>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculator-container"
    >
      <h2>
        Lets Do Some Math !!
      </h2>
      <table
        class="calculator"
      >
        <tbody>
          <tr>
            <th
              class="th-result"
              colspan="4"
            >
              <p
                class="calculator-result"
                data-testid="result"
              >
                0
              </p>
            </th>
          </tr>
          <tr>
            <th
              class="w-25"
            >
              <button
                class="btn-Gray"
                type="button"
              >
                AC
              </button>
            </th>
            <th
              class="w-25"
            >
              <button
                class="btn-Gray"
                type="button"
              >
                +/-
              </button>
            </th>
            <th
              class="w-25"
            >
              <button
                class="btn-Gray"
                type="button"
              >
                %
              </button>
            </th>
            <th
              class="w-25"
            >
              <button
                class="btn-Orange"
                type="button"
              >
                รท
              </button>
            </th>
          </tr>
          <tr>
            <th>
              <button
                class="btn-Gray"
                type="button"
              >
                7
              </button>
            </th>
            <th>
              <button
                class="btn-Gray"
                type="button"
              >
                8
              </button>
            </th>
            <th>
              <button
                class="btn-Gray"
                type="button"
              >
                9
              </button>
            </th>
            <th>
              <button
                class="btn-Orange"
                type="button"
              >
                x
              </button>
            </th>
          </tr>
          <tr>
            <th>
              <button
                class="btn-Gray"
                type="button"
              >
                4
              </button>
            </th>
            <th>
              <button
                class="btn-Gray"
                type="button"
              >
                5
              </button>
            </th>
            <th>
              <button
                class="btn-Gray"
                type="button"
              >
                6
              </button>
            </th>
            <th>
              <button
                class="btn-Orange"
                type="button"
              >
                +
              </button>
            </th>
          </tr>
          <tr>
            <th>
              <button
                class="btn-Gray"
                type="button"
              >
                1
              </button>
            </th>
            <th>
              <button
                class="btn-Gray"
                type="button"
              >
                2
              </button>
            </th>
            <th>
              <button
                class="btn-Gray"
                type="button"
              >
                3
              </button>
            </th>
            <th>
              <button
                class="btn-Orange"
                type="button"
              >
                -
              </button>
            </th>
          </tr>
          <tr>
            <th
              colspan="2"
            >
              <button
                class="btn-Gray"
                type="button"
              >
                0
              </button>
            </th>
            <th>
              <button
                class="btn-Gray"
                type="button"
              >
                .
              </button>
            </th>
            <th>
              <button
                class="btn-Orange"
                type="button"
              >
                =
              </button>
            </th>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
